acceptance/tests/peering/peering_connect_test.go

```diff
@@ -0,0 +1,330 @@
+				helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
+					k8s.KubectlDeleteK(t, staticServerPeerClusterContext.KubectlOptions(t), "../fixtures/cases/crd-peers/default-namespace")
+				})
+			} else {
+				k8s.KubectlApplyK(t, staticServerPeerClusterContext.KubectlOptions(t), "../fixtures/cases/crd-peers/non-default-namespace")
+				helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
+					k8s.KubectlDeleteK(t, staticServerPeerClusterContext.KubectlOptions(t), "../fixtures/cases/crd-peers/non-default-namespace")
+				})
+			}
+
+			logger.Log(t, "checking that connection is successful")
+			k8s.CheckStaticServerConnectionSuccessful(t, staticClientOpts, staticClientName, "http://localhost:1234")
+
+			denyAllIntention := &api.ServiceIntentionsConfigEntry{
+				Name:      "*",
+				Kind:      api.ServiceIntentions,
+				Namespace: "*",
+				Sources: []*api.SourceIntention{
+					{
+						Name:      "*",
+						Namespace: "*",
+						Action:    api.IntentionActionDeny,
+						Peer:      staticClientPeer,
+					},
+				},
+			}
+			_, _, err = staticServerPeerClient.ConfigEntries().Set(denyAllIntention, &api.WriteOptions{})
+			require.NoError(t, err)
+
+			helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
+				_, err := staticServerPeerClient.ConfigEntries().Delete(api.ServiceIntentions, "*", &api.WriteOptions{})
+				require.NoError(t, err)
+			})
+
+			logger.Log(t, "checking that the connection is not successful because there's no allow intention")
+			k8s.CheckStaticServerConnectionFailing(t, staticClientOpts, staticClientName, "http://localhost:1234")
+
+			intention := &api.ServiceIntentionsConfigEntry{
+				Name:      staticServerName,
+				Kind:      api.ServiceIntentions,
+				Namespace: staticServerNamespace,
+				Sources: []*api.SourceIntention{
+					{
+						Name:      staticClientName,
+						Namespace: staticClientNamespace,
+						Action:    api.IntentionActionAllow,
+						Peer:      staticClientPeer,
+					},
+				},
+			}
+
+			// Set the destination namespace to be the same
+			// unless mirrorK8S is true.
+			if !c.mirrorK8S {
+				intention.Namespace = c.destinationNamespace
+				intention.Sources[0].Namespace = c.destinationNamespace
+			}
+
+			logger.Log(t, "creating intentions in server peer")
+			_, _, err = staticServerPeerClient.ConfigEntries().Set(intention, &api.WriteOptions{})
+			require.NoError(t, err)
+
+			helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
+				_, err := staticServerPeerClient.ConfigEntries().Delete(api.ServiceIntentions, staticServerName, &api.WriteOptions{})
+				require.NoError(t, err)
+			})
+
+			logger.Log(t, "checking that connection is successful")
+			k8s.CheckStaticServerConnectionSuccessful(t, staticClientOpts, staticClientName, "http://localhost:1234")
+		})
+	}
+}
+
+func copySecret(t *testing.T, cfg *config.TestConfig, sourceContext, destContext environment.TestContext, secretName string) {
```

Comment:
we could move this function to helpers because I think we now use this functionality in a few places (mesh gw tests and partition tests)